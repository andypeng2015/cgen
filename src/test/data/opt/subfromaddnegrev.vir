module foo

export function w $foo(w %x, w %y) {
@start:
  %y = neg.w %y
  %x = add.w %y, %x
  ret %x
}
