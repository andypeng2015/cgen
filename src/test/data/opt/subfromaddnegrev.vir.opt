module foo

export function w $foo(w %x, w %y) {
@2:
  %1 = sub.w %x, %y ; @6
  ret %1
}
