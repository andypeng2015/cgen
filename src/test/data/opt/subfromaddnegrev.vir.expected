module foo

export function w $foo(w %x, w %y) {
@2:
  %0 = sub.w %x, %y ; @5
  ret %0
}
